thunder/core/jit_ext.py

```diff
@@ -1402,7 +1427,7 @@ def from_binary_subscr(provenance, *, new_output=False):
                 output = Proxy("subscr")  # name? collectify?
             else:
                 output = p
-            if isinstance(idx, (int, str)):
+            if isinstance(idx, (int, str, Proxy)):
```

Comment:
Patch for https://github.com/Lightning-AI/lightning-thunder/issues/1266 (shouldn't be part of this PR)